pkg/cable/libreswan/libreswan.go

```diff
@@ -616,28 +616,41 @@ func (i *libreswan) runPluto() error {
 		logger.Fatalf("Pluto exited: %v", cmd.Wait())
 	}()
 
-	// Wait up to 5s for the control socket.
-	for i := 0; i < 250; i++ {
-		_, err := os.Stat("/run/pluto/pluto.ctl")
+	err := i.waitForControlSocket()
+	if err != nil {
+		return err
+	}
+
+	if i.debug {
+		if err := whack("--debug", "base"); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (i *libreswan) waitForControlSocket() error {
+	// Wait for upto a minute for the control socket to be created.
+	const maxAttempts = 600
+	const retryInterval = 100 * time.Millisecond
```

Comment:
QQ: don't we want to monitor the status at a faster rate? to shorten DP downtime when GW pod is restarted ?

Reply:
I can reduce it to 20 ms like before, but it also means more API calls. Also, looking at the time it normally takes on OCP which is in seconds, 100 ms sounded a reasonable time for me. Hence I used 100 ms.

Reply:
OK, let's leave it 100mSec, we can optimize the interval in the future